ONSdigital/response-operations-ui
# Link collection instrument

# Motivation and Context
Previously we'd have to use `curl` commands to link collection instruments to surveys.  This was a bit risky to do in preprod and prod as the url isn't exactly simple.

# What has changed
- New `/surveys/link-collection-instrument` page created and template added
- `/surveys/create` code and template modified so it uses the values in the form so each value doesn't need to be added in render_template.

# How to test?
- Go to `/surveys/link-collection-instrument` and try it out. Good things to test would be empty fields, a survey that doesn't exist and modifying the http authentication details for the collection service to see it fail in various ways.

The errors are a bit varied (some spit out json, others a text string), but the aim for it was to be useful to whoever is doing the work.  It currently gives you the ugly 500 error page if either the survey service or collection instrument service aren't up.  It's a bit jarring but there's not really a graceful way to handle it without writing more code for something which almost never happens

# Links
https://trello.com/c/JA9rHJBk

# Screenshots (if appropriate):
